### PR TITLE
WIP: added i18n using Spanish as example

### PR DESCRIPTION
**Pull request by mkobayashi21**
build, taking away all buttons to change language except in homepage
first steps/WIP to #42

### CODE REVIEW

File: src/components/RecordForm/index.js [hunk 1]
```diff
@@ -249,6 +251,8 @@ class RecordForm extends Component {
 
 	render(){
 		let isSelected = (this.props.selectedRecord && this.state.recordID === this.props.selectedRecord["o:id"]);
+		const strings = this.state.strings;
```
Comment (akstuhl) [inline src/components/RecordForm/index.js]: This line should be unneeded.

File: src/components/exhibitForm.js [hunk 2]
```diff
@@ -37,6 +38,8 @@ class ExhibitForm extends Component {
 
 	constructor(props) {
 		super(props);
+		//i18n
+		this.state = {strings};
```
Comment (akstuhl) [inline src/components/exhibitForm.js]: Unneeded (see review summary comment)

File: src/components/exhibitForm.js [hunk 3]
```diff
@@ -59,30 +62,34 @@ class ExhibitForm extends Component {
 		this.props.dispatch(this.set_availableTileLayers({ids: arrayOfIDs}));
 	}
 	render() {
+		const strings = this.state.strings;
```
Comment (akstuhl) [inline src/components/exhibitForm.js]: Unneeded (see review summary comment)

File: src/views/exhibits/index.js [hunk 4]
```diff
@@ -6,33 +6,41 @@ import { connect } from 'react-redux';
 import { fetchExhibits } from '../../reducers/not_refactored/exhibits';
 import { resetExhibit } from '../../reducers/not_refactored/exhibitShow';
 import { deleteExhibit } from '../../reducers/not_refactored/exhibitDelete';
+import { strings } from '../../i18nLibrary';
 
 class Exhibits extends Component {
   componentWillMount() {
     this.props.fetchExhibits();
     this.props.resetExhibit();
+    this.state = {strings};
```
Comment (akstuhl) [inline src/views/exhibits/index.js]: Unneeded (see review summary comment)

File: src/views/exhibits/index.js [hunk 5]
```diff
@@ -6,33 +6,41 @@ import { connect } from 'react-redux';
...
   }
 
   render() {
     const props = this.props;
-
+    const strings = this.state.strings;
```
Comment (akstuhl) [inline src/views/exhibits/index.js]: Unneeded (see review summary comment)

File: src/views/exhibits/show.js [hunk 6]
```diff
@@ -43,6 +45,7 @@ const RecordEditor = props => {
 class ExhibitShow extends Component {
 	componentWillMount() {
 		this.props.fetchExhibitWithRecords(this.props.match.params.slug);
+		this.state = {strings};
```
Comment (akstuhl) [inline src/views/exhibits/show.js]: Unneeded (see review summary comment)

File: src/views/exhibits/show.js [hunk 7]
```diff
@@ -58,11 +61,13 @@ class ExhibitShow extends Component {
 	render() {
 		const props = this.props;
 		const {exhibit} = props;
+		const strings = this.state.strings;
```
Comment (akstuhl) [inline src/views/exhibits/show.js]: Unneeded (see review summary comment)

**Review by akstuhl — CHANGES_REQUESTED** [review]
Hi @mkobayashi21, excellent work on this. The language toggling works seamlessly and the syntax is very straightforward. react-localization seems like a good fit for this project based on your implementation. One change needs to be made before you merge: in some of the components, there are lines setting/getting the `strings` variable in the react `state` object. This appears to be unnecessary based on my testing (`strings` is already available to the component via your import statement), and will generate a console error because react tries to prevent you from modifying the `state` object directly. I think I flagged all the lines that should be removed with inline comments here. If you can try removing those and confirm everything still displays correctly, this will be in great shape. Thanks!

**Comment by mkobayashi21** [comment]
Should be fine now; thank you @akstuhl  for reviewing!

**Review by akstuhl — APPROVED** [review]

File: src/i18nLibrary.js [hunk 8]
```diff
@@ -0,0 +1,233 @@
+import React, {Component} from 'react';
+import LocalizedStrings from 'react-localization';
+
+export let strings = new LocalizedStrings({
+ es:{
+   //PLEASE note: Spanish translations are very rough, and they only exist
+   //to demonstrate the i18n functionality, with no intent to offend anyone.
```
Comment (jamiefolsom) [inline src/i18nLibrary.js]: Love it

File: src/i18nLibrary.js [hunk 9]
```diff
@@ -0,0 +1,233 @@
+import React, {Component} from 'react';
+import LocalizedStrings from 'react-localization';
+
+export let strings = new LocalizedStrings({
```
Comment (jamiefolsom) [inline src/i18nLibrary.js:4]: Is there an easy way to export (or factor out) the strings as a text file, to be worked on by translators, then re-imported, pre-build?

File: src/views/exhibits/index.js [hunk 10]
```diff
@@ -6,33 +6,41 @@ import { connect } from 'react-redux';
...
+  
+    const changeLanguage = lng => {
+      strings.setLanguage(lng);
+      this.setState({});
+    }
     return (
       <div>
-        <h3><Link to={`${window.baseRoute}/`}>Neatline</Link> | Browse Exhibits</h3>
+        <button onClick={() => changeLanguage("sp")}>sp</button>
+        <button onClick={() => changeLanguage("en")}>en</button>
```
Comment (jamiefolsom) [inline src/views/exhibits/index.js]: Idea for discussion: could we detect available languages, and add buttons for each?

**Review by jamiefolsom — APPROVED** [review]
Looks good to me 👍 🎉

**Comment by jamiefolsom** [comment]
@mkobayashi21 if you would, rebase against master, and squash all but the first commit; if we can keep this PR open, and rebased on master for now, we might wait for @ZoeLeBlanc to review it.

**Comment by mkobayashi21** [comment]
Thanks @jamiefolsom ! It should be rebased/squashed now, and I will look into the questions that you raised.

**Comment by jamiefolsom** [comment]
Trying to reopen.

**Pull request merged**